src/crimson/os/seastore/extentmap_manager.h

```diff
@@ -0,0 +1,119 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <iostream>
+
+#include <boost/intrusive_ptr.hpp>
+#include <boost/smart_ptr/intrusive_ref_counter.hpp>
+
+#include <seastar/core/future.hh>
+
+#include "common/Formatter.h"
+#include "include/ceph_assert.h"
+#include "include/buffer_fwd.h"
+
+#include "crimson/osd/exceptions.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+
+#define PAGE_SIZE 4096
+#define EXTMAP_BLOCK_SIZE 4096
+
+namespace crimson::os::seastore {
+
+struct lext_map_val_t {
+  laddr_t laddr;
+  extent_len_t length = 0;
+
+  lext_map_val_t(
+    laddr_t laddr,
+    extent_len_t length)
+    : laddr(laddr), length(length) {}
+
+};
+class Extent
+{
+public:
+  objaddr_t logical_offset = 0;  //offset in object
+  laddr_t laddr;     // lextent start address aligned with block size.
+  extent_len_t length = 0;
+  explicit Extent(objaddr_t lo) : logical_offset(lo) { }
+
+  Extent(
+    objaddr_t lo,
+    laddr_t laddr,
+    extent_len_t length)
+    : logical_offset(lo), laddr(laddr), length(length) {}
+
+  void dump(ceph::Formatter* f) const;
+    friend bool operator<(const Extent &a, const Extent &b) {
+      return a.logical_offset < b.logical_offset;
+    }
+    friend bool operator>(const Extent &a, const Extent &b) {
+      return a.logical_offset > b.logical_offset;
+    }
+    friend bool operator==(const Extent &a, const Extent &b) {
+      return a.logical_offset == b.logical_offset;
+    }
+
+  ~Extent() {}
+};
+
+using ExtentRef = std::unique_ptr<Extent>;
+using extent_map_list_t = std::list<ExtentRef>;
+std::ostream &operator<<(std::ostream &out, const Extent &rhs);
+std::ostream &operator<<(std::ostream &out, const extent_map_list_t &rhs);
+
+struct extmap_root_t {
+  depth_t depth = 0;
+  laddr_t extmap_root_laddr;
+  extmap_root_t(depth_t dep, laddr_t laddr)
+  : depth(dep),
+    extmap_root_laddr(laddr) {}
+};
+
+using extmap_root_ref = std::shared_ptr<extmap_root_t>;
+
+/**
+ * Abstract interface for managing the object inner offset to logical addr mapping
+ * each onode has an extentmap tree for a particular onode.
+ */
+class ExtentMapManager {
+public:
+  using alloc_extmap_root_ertr = TransactionManager::alloc_extent_ertr;
+  using alloc_extmap_root_ret = alloc_extmap_root_ertr::future<>;
+  virtual alloc_extmap_root_ret alloc_extmap_root(Transaction &t) = 0;
```

Comment:
This is meant to be an abstract interface, right?  root implies a tree.  Maybe just call it initialize?  Also, shouldn't it return the extmap_root_t so that the user can record the addr/depth to access it again in the future?